gio/src/inet_socket_address.rs

```diff
@@ -0,0 +1,24 @@
+use InetAddress;
+use InetSocketAddress;
+use InetSocketAddressExt;
+
+use std::net::SocketAddr;
+
+impl InetSocketAddress {
+    pub fn from_socket_address(addr: SocketAddr) -> Self {
+        InetSocketAddress::new::<InetAddress>(&addr.ip().into(), addr.port())
+    }
+}
+
+impl From<SocketAddr> for InetSocketAddress {
+    fn from(addr: SocketAddr) -> Self {
+        InetSocketAddress::from_socket_address(addr)
+    }
+}
+
+impl Into<SocketAddr> for InetSocketAddress {
```

Comment:
```suggestion
impl From<InetSocketAddress> for SocketAddr {
```

Would be better and also possible, or not?

Reply:
I think that's better :). Let me update.

Reply:
I've updated it for everything else here: https://github.com/gtk-rs/gtk-rs/pull/136